models.py

```diff
@@ -1,209 +1,201 @@
-        """Return the Tag as a dictionary."""
-        return {"tag_type": self.tag_type,
-                "value": self.value}
-
-    def as_json(self):
-        """Return the Tag as a JSON string."""
-        return json.dumps(self.as_dict())
+    def __init__(self, uni_a, uni_z, name, start_date=None, end_date=None,
+                 bandwidth=None, primary_links=None, backup_links=None,
+                 dynamic_backup_path=None, creation_time=None):
+        """Create an EVC instance with the provided parameters.
 
+        Do some basic validations to attributes.
+        """
 
-class Endpoint:
-    """Class that represents an endpoint according to MEF 6.2 and MEF 10.3."""
+        if uni_a is None or uni_z is None or name is None:
+            raise TypeError("Invalid arguments")
 
-    def __init__(self, dpid, port, tag=None):
-        """Create an Endpoint instance and set its attributes."""
-        self.dpid = str(dpid)
-        self.port = str(port)
-        self.tag = tag or Tag()
+        if ((not isinstance(uni_a, UNI)) or
+                (not isinstance(uni_z, UNI))):
+            raise TypeError("Invalid UNI")
 
-    def __eq__(self, other):
-        """Check if two instances of Endpoint are equal."""
-        return self.dpid == other.dpid and self.port == other.port
+        if not uni_a.is_valid() or not uni_z.is_valid():
+            raise TypeError("Invalid UNI")
 
-    def is_valid(self):
-        """Check if an Endpoint has valid properties."""
-        if not isinstance(self.dpid, str):
+        self._id = uuid4().hex
+        self.uni_a = uni_a
+        self.uni_z = uni_z
+        self.name = name
+        self.start_date = start_date if start_date else now()
+        self.end_date = end_date
+        # Bandwidth profile
+        self.bandwidth = bandwidth
+        self.primary_links = primary_links
+        self.backup_links = backup_links
+        self.dynamic_backup_path = dynamic_backup_path
+        # dict with the user original request (input)
+        self._requested = None
+        # circuit being used at the moment if this is an active circuit
+        self.current_path = None
+        # primary circuit offered to user IF one or more links were provided in
+        # the request
+        self.primary_path = None
+        # backup circuit offered to the user IF one or more links were provided
+        # in the request
+        self.backup_path = None
+        # datetime of user request for a EVC (or datetime when object was
+        # created)
+        self.request_time = now()
+        # datetime when the circuit should be activated. now() || schedule()
+        self.creation_time =  creation_time or now()
+        self.owner = None
+        # Operational State
+        self.active = False
+        # Administrative State
+        self.enabled = False
+        # Service level provided in the request. "Gold", "Silver", ...
+        self.priority = 0
+        # (...) everything else from request must be @property
```

Comment:
Maybe those comments about the attributes would be better located in the docstring.